components/net/http_loader.rs

```diff
@@ -1455,17 +1455,18 @@ fn cors_preflight_fetch(
     context: &FetchContext,
 ) -> Response {
     // Step 1
-    let mut preflight = Request::new(
-        request.current_url(),
-        Some(request.origin.clone()),
-        request.pipeline_id,
-    );
-    preflight.method = Method::OPTIONS;
-    preflight.initiator = request.initiator.clone();
-    preflight.destination = request.destination.clone();
-    preflight.origin = request.origin.clone();
-    preflight.referrer = request.referrer.clone();
-    preflight.referrer_policy = request.referrer_policy;
+    let mut preflight = RequestBuilder::new(request.current_url())
+        .method(Method::OPTIONS)
+        .origin(match request.origin.clone() {
+            Origin::Client => ImmutableOrigin::new_opaque(),
+            Origin::Origin(origin) => origin,
```

Comment:
This is a behavior change, can you explain why this is necessary?

Reply:
I'd probably also write it as:
```rust
.origin(match request.origin {
  Origin::Client =>  ImmutableOrigin::new_opaque(),
  Origin::Origin(origin) => origin.clone(),
})
```

Reply:
Sorry for the late reply!

> can you explain why this is necessary

https://github.com/servo/servo/blob/a8ae0e56dd75e6467e063c7f8014ab94ba07f0a0/components/net_traits/request.rs#L260-L263

The `origin` function in builder needs an immutable origin so I tried to do this mapping in this function.
Would you think it's better to have a function to do this? 🤔

> I'd probably also write it as

Ok! Will update! I think the benefit here is we don't need to do a redundant `clone` in the client case, right?

Reply:
I'm concerned that this could silently be a change in behaviour. I would rather have the Client branch use unimplemented!() to be sure we notice if we're accidentally hitting it.

Reply:
> have the Client branch use unimplemented!()

ok! I will try to use `unimplemented!` macro here 🤔

I'm wondering if it's worth having a test case for this function? 🤔 so that we can catch behavior changes earlier? 👀

Reply:
A testcase would require a request that uses a Client origin that also triggers a preflight request, and would verify that both of the requests provide the same Origin header.

Reply:
Updated to use `unimplemented!` macro but not adding any tests yet